pkg/deploy/controller/deploymentconfig/factory.go

```diff
@@ -81,3 +81,259 @@ func (factory *DeploymentConfigControllerFactory) Create() controller.RunnableCo
 		},
 	}
 }
+
+const (
+	// FullDeploymentConfigResyncPeriod means we'll attempt to reconcile deploymentconfigs
+	// every two minutes.
+	FullDeploymentConfigResyncPeriod = 2 * time.Minute
+	// We must avoid creating new replication controllers until the {replication controller,pods} store
+	// has synced. If it hasn't synced, to avoid a hot loop, we'll wait this long between checks.
+	StoreSyncedPollPeriod = 100 * time.Millisecond
+)
+
+// NewDeploymentConfigController creates a new DeploymentConfigController.
+func NewDeploymentConfigController(oc osclient.Interface, kc kclient.Interface, codec runtime.Codec) *DeploymentConfigController {
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartRecordingToSink(kc.Events(""))
+	recorder := eventBroadcaster.NewRecorder(kapi.EventSource{Component: "deploymentconfig-controller"})
+
+	c := &DeploymentConfigController{
+		kc: kc,
+		oc: oc,
+
+		// TODO: Use ratelimiting queue once the rebase lands
```

Comment:
Fixed up before it merged :)
https://github.com/openshift/origin/blob/master/pkg/service/controller/servingcert/controller.go#L53

On Thu, May 26, 2016 at 5:47 AM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/deploy/controller/deploymentconfig/factory.go
> <https://github.com/openshift/origin/pull/9002#discussion_r64718646>:
>
> > +	// We must avoid creating new replication controllers until the {replication controller,pods} store
> > +	// has synced. If it hasn't synced, to avoid a hot loop, we'll wait this long between checks.
> > +	StoreSyncedPollPeriod = 100 * time.Millisecond
> > +)
> > +
> > +// NewDeploymentConfigController creates a new DeploymentConfigController.
> > +func NewDeploymentConfigController(oc osclient.Interface, kc kclient.Interface, codec runtime.Codec) *DeploymentConfigController {
> > +	eventBroadcaster := record.NewBroadcaster()
> > +	eventBroadcaster.StartRecordingToSink(kc.Events(""))
> > +	recorder := eventBroadcaster.NewRecorder(kapi.EventSource{Component: "deploymentconfig-controller"})
> > +
> > +	c := &DeploymentConfigController{
> > +		kc: kc,
> > +		oc: oc,
> > +
> > +		// TODO: Use ratelimiting queue once the rebase lands
>
> right, was in the service cert controller *I* reviewed...
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/9002/files/df59642f97cc27380b70126d7cbd49032e8e7223#r64718646>
>